### DISCUSSION TITLE
What`s the classification standard of the hydrogeological data

### DISCUSSION THREAD

**Discussion by LuvMangorange** [mhm-ufz/mHM#22] [Q&A] [answered]
Hello !
I am trying to use the mHM model to simulate the SM on the WeiHe watershed in China. According to the manual, I have completed other data in my sample except 'geology_class.asc' and  'geology_classdefinition.txt' .

These questions  may need to be resolved urgently, as the bgr wetsite does not provide the hydrogeological map of China:
1) Is the hydrogeological data /  'geology_class.asca, geology_classdefinition.txt'  mandatory？
2) What is the classification standard of the hydrogeological data？

Note: mhm version is mhm-v5.11.1.

**Comment by pallavhydro** [comment]
Hi there

Great that you have almost completed the mhm setup. Very impressive!

My replies -
1. Yes.
2. There is classification standard so as to say. mhm just needs classes. Based on the classes (map - `geology_class.asc`, look up table (LUT) - `geology_classdefinition.txt`), and the parameters (in geology section of `mhm_parameters.nml`), mhm achieves spatial variation of recession constants throughout the domain. 

For the geology data, following would be the `procedure`:
- Convert your geological map to map with integer classes
- The number of classes depends on the grouping resolution you want to achieve. But beware that larger the number of classes, more will be the number of parameters in `mhm_parameters.nml`. Just to give you an idea, the test domains of mhm have only 2 classes while in some of our other projects we have used up to 16 classes. 
- Prepare the corresponding `geology_classdefinition.txt`. If you don't have the karstic info, keep it 0.
- Adjust the number of geological parameters in `mhm_parameters.nml` to match the number of classes.

I hope this helps. Let us know if you succeed/ stumble across any issues.

Pallav

**Reply by LuvMangorange** [reply]
Thank you very much,！
But the problem still exists. 
How to classify the map， or  What is my reason for classifying the map?

**Reply by rohinikumar82** [reply]
Dear @LuvMangorange,

As I wrote below - the classification would depend on the granularity with which you want to capture for the hydrogeological features of your catchment; as well what the underlying data would support. In many cases, the hydrogeological map provide some categorical classes (choose the main/domiannt features that mostly related to flow characteristics). Note that in mHM the groundwater features are not yet fully regionalised in ways that other parts are done. We are in process of coupling the mHM (surface/near-surface features) with a detail GW model in this respect.  
Stay tuned!
Best,
Rohini

**Reply by LuvMangorange** [reply]
Thanks again!
😀The answer I want has actually appeared in your answer--"mhm realizes the spatial change of the decay constant in the entire domain." ，when I read your reply again.

**Comment by rohinikumar82** [comment, marked as answer]
Dear @LuvMangorange,

At the moment - in mHM - geological parameter, recession constant (k2) affecting base flows - is not regionalised as other parameters i.e., linking k2 to geological characteristics in some continuous functional form - rather it uses categorical classification based approach. 
(As Pallav indicated above) you can: 
- the (hydro) geological map of the study area - I guess this is usually in some shape file
- identify some dominant geological classes (say 2-10 depending on the region) and give them some unique ids 
- convert the file in raster and then to ASCII format at the L0 resolution (similar to DEM one) and make "geology_class.asc"
- change the # of geological related parameters in the mhm_parameter.nml file: you can use the same parameter range and default values for all classes (as given now)
- change also the geology look-up table file (geology_classdefinition.tx) accordingly. In geology LUT file you have option to open/close the Karrstic aquifer related options - if you know in your area there is a karstic aquifers, otherwise you can leave it out as it is in default case.

Is the hydrogeological data / 'geology_class.asca, geology_classdefinition.txt' mandatory？- yes you need these files to run mHM. 

Hope these helps. Happy coding/running mHM

Best,
Rohini

**Reply by LuvMangorange** [reply]
Great！Thank you for the specific steps.
I think I have understood the meaning of the hydrogeological map in mHM model.
Thanks！
